### ISSUE TITLE
Operations defined by structured header should support an aoid override

### ISSUE THREAD

**Issue by gibson042** [tc39/ecmarkup#398]
The Temporal proposal has a need to mention both the 262 and 402 flavors of some operations, which ecmarkup rejects as duplicate definitions. This can be addressed for traditionally-defined operations via an aoid suffix, but an attempt to do the same for a structured operation is rejected ("_nodes with structured headers should not include an AOID_"). I'm attempting to work around this in https://github.com/tc39/proposal-temporal/pull/1920/commits/82f542b2765314767577c0ec4a5a65b1326a7afc by effectively suppressing the structured definition, but it would be better if instead something like `<emu-clause id="sup-canonicalizetimezonename" type="abstract operation" aoid="CanonicalizeTimeZoneName402">` were supported.

**Comment by bakkot** [comment]
Can you say more about why you need these to be AOIDs?

**Comment by gibson042** [comment]
They don't need to be AOIDs specifically, that just seemed like an obvious option.

**Comment by gibson042** [comment]
I am trying to have two abstract operations in the same proposal spec which are both defined by structured header including the same name (one for showing changes in ECMA-262, the other for showing changes in ECMA-402) but _do not result in an error from `ecmarkup --lint-spec --strict`_, ideally because ecmarkup has a way to recognize them as distinct despite that identical name. This is not currently possible with structured headers, but _is_ possible without them:
```html
<emu-clause id="sec-temporal-isbuiltincalendar" aoid="IsBuiltinCalendar">
  <h1>IsBuiltinCalendar ( _id_ )</h1>
```
```html
<emu-clause id="sup-temporal-isbuiltincalendar" aoid="IsBuiltinCalendar402">
  <h1>IsBuiltinCalendar ( _id_ )</h1>
```

**Comment by bakkot** [comment]
Ah, ok. And to be clear, in the snippet you give without structured headers, you're not actually using the AOID for anything, right? That is, you don't expect to be able to write `IsBuiltinCalendar402("foo")`  in an algorithm step and have it link to that second definition?

The solution I'm currently envisioning is a field in the `dl` along the lines of `<dt>redefinition</dt><dd>true</dd>`, which if present would suppress that warning.

Such a field could also cause ecmarkup to
- automatically add a note to the original section which links to the other definition(s)
- check that the signatures are compatible

unless you're opposed.

But this solution would not have the effect of allowing you to refer to it and have the reference automatically link, like happens with `IsBuiltinCalendar402` in the snippet you've given.

**Comment by gibson042** [comment]
> Ah, ok. And to be clear, in the snippet you give without structured headers, you're not actually using the AOID for anything, right? That is, you don't expect to be able to write `IsBuiltinCalendar402("foo")` in an algorithm step and have it link to that second definition?

Correct; the ECMA-402 version is _technically_ referenced from other ECMA-402 operations that do appear in the proposal, but it's acceptable for those references to instead jump to the ECMA-262 version (e.g., like the IsValidTimeZoneName links in https://tc39.es/proposal-temporal/#sup-canonicalizetimezonename go to [#sec-isvalidtimezonename](https://tc39.es/proposal-temporal/#sec-isvalidtimezonename) rather than the immediately preceding [#sup-isvalidtimezonename](https://tc39.es/proposal-temporal/#sup-isvalidtimezonename)) because the target audience can be trusted to understand.

> The solution I'm currently envisioning is a field in the `dl` along the lines of `<dt>redefinition</dt><dd>true</dd>`, which if present would suppress that warning.
> 
> Such a field could also cause ecmarkup to
> 
> * automatically add a note to the original section which links to the other definition(s)
> * check that the signatures are compatible
> 
> unless you're opposed.
> 
> But this solution would not have the effect of allowing you to refer to it and have the reference automatically link, like happens with `IsBuiltinCalendar402` in the snippet you've given.

Thanks, works for me!
